--- origsrc/cmake-2.4.8/Modules/FindFLTK.cmake	2008-01-21 12:59:51.000000000 -0600
+++ src/cmake-2.4.8/Modules/FindFLTK.cmake	2008-07-29 00:24:05.015625000 -0500
@@ -31,10 +31,6 @@
   SET( FLTK_PLATFORM_DEPENDENT_LIBS  "-framework Carbon -framework Cocoa -framework ApplicationServices -lz")
 ENDIF(APPLE)
 
-IF(CYGWIN)
-  SET( FLTK_PLATFORM_DEPENDENT_LIBS ole32 uuid comctl32 wsock32 supc++ -lm -lgdi32)
-ENDIF(CYGWIN)
-
 # If FLTK_INCLUDE_DIR is already defined we assigne its value to FLTK_DIR
 IF(FLTK_INCLUDE_DIR)
   SET(FLTK_DIR ${FLTK_INCLUDE_DIR})
