--- origsrc/cmake-2.6.1/Modules/Platform/CYGWIN.cmake	2008-08-01 10:34:50.000000000 -0500
+++ src/cmake-2.6.1/Modules/Platform/CYGWIN.cmake	2008-08-27 22:26:25.905254600 -0500
@@ -1,25 +1,25 @@
-SET(WIN32 1)
+SET(WIN32 0)
 SET(CYGWIN 1)
 
-SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-shared -Wl,--export-all-symbols -Wl,--enable-auto-import")
+SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-shared -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--enable-auto-image-base")
 SET(CMAKE_SHARED_MODULE_CREATE_C_FLAGS ${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS})
-SET(CMAKE_DL_LIBS "-lgdi32" )
+SET(CMAKE_DL_LIBS "" )
 SET(CMAKE_SHARED_LIBRARY_PREFIX "cyg")
 SET(CMAKE_SHARED_LIBRARY_SUFFIX ".dll")
-SET(CMAKE_SHARED_MODULE_PREFIX "lib")
+SET(CMAKE_SHARED_MODULE_PREFIX "cyg")
 SET(CMAKE_SHARED_MODULE_SUFFIX ".dll")
 SET(CMAKE_IMPORT_LIBRARY_PREFIX "lib")
 SET(CMAKE_IMPORT_LIBRARY_SUFFIX ".dll.a")
 # no pic for gcc on cygwin
-SET(CMAKE_SHARED_LIBRARY_C_FLAGS "")
-SET(CMAKE_SHARED_LIBRARY_CXX_FLAGS "")
+SET(CMAKE_SHARED_LIBRARY_C_FLAGS "-DDLL_EXPORT")
+SET(CMAKE_SHARED_LIBRARY_CXX_FLAGS "-DDLL_EXPORT")
 SET(CMAKE_EXECUTABLE_SUFFIX ".exe")          # .exe
 SET(CMAKE_CREATE_WIN32_EXE  "-mwindows")
 # Modules have a different default prefix that shared libs.
 SET(CMAKE_MODULE_EXISTS 1)
 
-SET(CMAKE_FIND_LIBRARY_PREFIXES "cyg" "lib")
-SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll" ".dll.a" ".a")
+SET(CMAKE_FIND_LIBRARY_PREFIXES "lib")
+SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll.a" ".a")
 
 SET(CMAKE_GNULD_IMAGE_VERSION
   "-Wl,--major-image-version,<TARGET_VERSION_MAJOR>,--minor-image-version,<TARGET_VERSION_MINOR>")
--- origsrc/cmake-2.6.1/Source/CMakeLists.txt	2008-08-01 10:34:51.000000000 -0500
+++ src/cmake-2.6.1/Source/CMakeLists.txt	2008-08-27 22:59:26.762974600 -0500
@@ -359,12 +359,12 @@
   CPack/cmCPackZIPGenerator.cxx
   )
 
-IF(WIN32 AND UNIX)
+IF(CYGWIN)
   SET(CPACK_SRCS ${CPACK_SRCS}
     CPack/cmCPackCygwinBinaryGenerator.cxx
     CPack/cmCPackCygwinSourceGenerator.cxx
     )
-ENDIF(WIN32 AND UNIX)
+ENDIF(CYGWIN)
   
 IF(UNIX)
   SET(CPACK_SRCS ${CPACK_SRCS}
--- origsrc/cmake-2.6.1/Source/WXDialog/bin/FindwxW.cmake	2008-08-01 10:34:55.000000000 -0500
+++ src/cmake-2.6.1/Source/WXDialog/bin/FindwxW.cmake	2008-08-27 22:30:25.264629600 -0500
@@ -247,10 +247,10 @@
     #
     # Find XWindows
     #
-    IF( NOT CYGWIN OR MINGW )
+    IF( NOT MINGW )
         INCLUDE( ${CMAKE_ROOT}/Modules/FindX11.cmake )
         SET( WXWIDGETS_INCLUDE_DIR ${WXWIDGETS_INCLUDE_DIR}  ${X11_INCLUDE_DIR} )
-    ENDIF( NOT CYGWIN OR MINGW )
+    ENDIF( NOT MINGW )
 ENDIF( WIN32_STYLE_FIND )
 
 #
@@ -384,9 +384,9 @@
 #
 SET( WXWIDGETS_RC ${WXWIDGETS_INSTALL_PATH}/include/wx/msw/wx.rc )
 IF( NOT WIN32_STYLE_FIND )
-    IF( NOT CYGWIN OR MINGW )
+    IF( NOT MINGW )
         SET( WXWIDGETS_RC "" )
-    ENDIF( NOT CYGWIN OR MINGW )
+    ENDIF( NOT MINGW )
 ENDIF( NOT WIN32_STYLE_FIND )
 
 #
@@ -397,10 +397,6 @@
 
 IF(WIN32_STYLE_FIND)
     SET( WXWIDGETS_DEFINITIONS ${WXWIDGETS_DEFINITIONS} -DWIN32 -D__WXMSW__  )
-ELSE(WIN32_STYLE_FIND)
-    IF( CYGWIN )
-        SET( WXWIDGETS_DEFINITIONS ${WXWIDGETS_DEFINITIONS} -D__GNUWIN32__  -O2 -D_WIN32_IE=0x400 -MMD -Wall  )
-    ENDIF( CYGWIN )
 ENDIF(WIN32_STYLE_FIND)
 
 IF( WXWIDGETS_USE_SHARED )
--- origsrc/cmake-2.6.1/Source/cmCacheManager.cxx	2008-08-01 10:34:51.000000000 -0500
+++ src/cmake-2.6.1/Source/cmCacheManager.cxx	2008-08-27 22:30:43.014629600 -0500
@@ -27,7 +27,7 @@
 
 #include <cmsys/RegularExpression.hxx>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 # include <windows.h>
 #endif // _WIN32
 
--- origsrc/cmake-2.6.1/Source/cmMakefile.cxx	2008-08-01 10:34:53.000000000 -0500
+++ src/cmake-2.6.1/Source/cmMakefile.cxx	2008-08-27 22:31:20.514629600 -0500
@@ -2216,18 +2216,13 @@
   working, these variables are still also set here in this place, but they
   will be reset in CMakeSystemSpecificInformation.cmake before the platform 
   files are executed. */
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
   this->AddDefinition("WIN32", "1");
   this->AddDefinition("CMAKE_HOST_WIN32", "1");
 #else
   this->AddDefinition("UNIX", "1");
   this->AddDefinition("CMAKE_HOST_UNIX", "1");
 #endif
-  // Cygwin is more like unix so enable the unix commands
-#if defined(__CYGWIN__)
-  this->AddDefinition("UNIX", "1");
-  this->AddDefinition("CMAKE_HOST_UNIX", "1");
-#endif
 #if defined(__APPLE__)
   this->AddDefinition("APPLE", "1");
   this->AddDefinition("CMAKE_HOST_APPLE", "1");
--- origsrc/cmake-2.6.1/Source/cmTarget.cxx	2008-08-01 10:34:53.000000000 -0500
+++ src/cmake-2.6.1/Source/cmTarget.cxx	2008-08-27 22:33:13.496333400 -0500
@@ -2439,8 +2439,13 @@
     if(type == cmTarget::SHARED_LIBRARY && !implib &&
        this->Makefile->IsOn("CMAKE_SHARED_LIBRARY_NAME_WITH_VERSION"))
       {
+      int major = 0;
+      char major_str [10];
+      sscanf(version, "%d.%*d", &major);
+      sprintf(major_str, "%d", major);
+
       outBase += "-";
-      outBase += version;
+      outBase += major_str;
       }
     }
 
--- origsrc/cmake-2.6.1/Source/kwsys/DynamicLoader.cxx	2008-08-01 10:34:54.000000000 -0500
+++ src/cmake-2.6.1/Source/kwsys/DynamicLoader.cxx	2008-08-27 22:33:47.041920200 -0500
@@ -74,7 +74,11 @@
 //----------------------------------------------------------------------------
 const char* DynamicLoader::LibPrefix()
 {
+#if defined(__CYGWIN__)
+  return "cyg";
+#else
   return "lib";
+#endif
 }
 
 //----------------------------------------------------------------------------
--- origsrc/cmake-2.6.1/Source/kwsys/SharedForward.h.in	2008-08-01 10:34:54.000000000 -0500
+++ src/cmake-2.6.1/Source/kwsys/SharedForward.h.in	2008-08-27 22:34:25.931051800 -0500
@@ -253,7 +253,7 @@
 #endif
 
 /* Windows */
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(__CYGWIN__)
 # if defined(__CYGWIN__)
 #  define KWSYS_SHARED_FORWARD_LDD "cygcheck"
 #  define KWSYS_SHARED_FORWARD_LDD_N 1
