--- origsrc/cmake-2.8.0/Utilities/cm_xmlrpc.h	2009-11-13 12:33:12.000000000 -0600
+++ src/cmake-2.8.0/Utilities/cm_xmlrpc.h	2009-11-24 22:01:15.145472700 -0600
@@ -15,8 +15,8 @@
 /* Use the xmlrpc library configured for CMake.  */
 #include "cmThirdParty.h"
 #ifdef CTEST_USE_XMLRPC
-# include <xmlrpc.h>
-# include <xmlrpc_client.h>
+# include <xmlrpc-c/base.h>
+# include <xmlrpc-c/client.h>
 #endif
 
 #endif
