--- origsrc/cmake-2.8.0/Modules/Platform/CYGWIN.cmake	2009-11-13 12:32:52.000000000 -0600
+++ src/cmake-2.8.0/Modules/Platform/CYGWIN.cmake	2009-11-25 02:04:37.577684100 -0600
@@ -1,26 +1,24 @@
-SET(WIN32 1)
+SET(WIN32 0)
 SET(CYGWIN 1)
 
-SET(CMAKE_EXE_LINKER_FLAGS_INIT "-Wl,--enable-auto-import")
-SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-shared -Wl,--export-all-symbols -Wl,--enable-auto-import")
+SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-shared")
 SET(CMAKE_SHARED_MODULE_CREATE_C_FLAGS ${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS})
-SET(CMAKE_DL_LIBS "-lgdi32" )
+SET(CMAKE_DL_LIBS "" )
 SET(CMAKE_SHARED_LIBRARY_PREFIX "cyg")
 SET(CMAKE_SHARED_LIBRARY_SUFFIX ".dll")
-SET(CMAKE_SHARED_MODULE_PREFIX "lib")
+SET(CMAKE_SHARED_MODULE_PREFIX "cyg")
 SET(CMAKE_SHARED_MODULE_SUFFIX ".dll")
 SET(CMAKE_IMPORT_LIBRARY_PREFIX "lib")
 SET(CMAKE_IMPORT_LIBRARY_SUFFIX ".dll.a")
 # no pic for gcc on cygwin
-SET(CMAKE_SHARED_LIBRARY_C_FLAGS "")
-SET(CMAKE_SHARED_LIBRARY_CXX_FLAGS "")
+SET(CMAKE_SHARED_LIBRARY_C_FLAGS "-DDLL_EXPORT")
+SET(CMAKE_SHARED_LIBRARY_CXX_FLAGS "-DDLL_EXPORT")
 SET(CMAKE_EXECUTABLE_SUFFIX ".exe")          # .exe
-SET(CMAKE_CREATE_WIN32_EXE  "-mwindows")
 # Modules have a different default prefix that shared libs.
 SET(CMAKE_MODULE_EXISTS 1)
 
-SET(CMAKE_FIND_LIBRARY_PREFIXES "cyg" "lib")
-SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll" ".dll.a" ".a")
+SET(CMAKE_FIND_LIBRARY_PREFIXES "lib")
+SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll.a" ".a")
 
 SET(CMAKE_GNULD_IMAGE_VERSION
   "-Wl,--major-image-version,<TARGET_VERSION_MAJOR>,--minor-image-version,<TARGET_VERSION_MINOR>")
--- origsrc/cmake-2.8.0/Source/CMakeLists.txt	2009-11-13 12:32:54.000000000 -0600
+++ src/cmake-2.8.0/Source/CMakeLists.txt	2009-11-24 22:11:27.524498800 -0600
@@ -417,12 +417,12 @@ SET(CPACK_SRCS
   CPack/cmCPackZIPGenerator.cxx
   )
 
-IF(WIN32 AND UNIX)
+IF(CYGWIN)
   SET(CPACK_SRCS ${CPACK_SRCS}
     CPack/cmCPackCygwinBinaryGenerator.cxx
     CPack/cmCPackCygwinSourceGenerator.cxx
     )
-ENDIF(WIN32 AND UNIX)
+ENDIF(CYGWIN)
   
 IF(UNIX)
   SET(CPACK_SRCS ${CPACK_SRCS}
--- origsrc/cmake-2.8.0/Source/cmCacheManager.cxx	2009-11-13 12:32:54.000000000 -0600
+++ src/cmake-2.8.0/Source/cmCacheManager.cxx	2009-11-24 22:11:27.547500100 -0600
@@ -22,7 +22,7 @@
 
 #include <cmsys/RegularExpression.hxx>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 # include <windows.h>
 #endif // _WIN32
 
--- origsrc/cmake-2.8.0/Source/cmMakefile.cxx	2009-11-13 12:32:57.000000000 -0600
+++ src/cmake-2.8.0/Source/cmMakefile.cxx	2009-11-24 22:11:27.576501800 -0600
@@ -2344,18 +2344,13 @@ void cmMakefile::AddDefaultDefinitions()
   working, these variables are still also set here in this place, but they
   will be reset in CMakeSystemSpecificInformation.cmake before the platform 
   files are executed. */
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
   this->AddDefinition("WIN32", "1");
   this->AddDefinition("CMAKE_HOST_WIN32", "1");
 #else
   this->AddDefinition("UNIX", "1");
   this->AddDefinition("CMAKE_HOST_UNIX", "1");
 #endif
-  // Cygwin is more like unix so enable the unix commands
-#if defined(__CYGWIN__)
-  this->AddDefinition("UNIX", "1");
-  this->AddDefinition("CMAKE_HOST_UNIX", "1");
-#endif
 #if defined(__APPLE__)
   this->AddDefinition("APPLE", "1");
   this->AddDefinition("CMAKE_HOST_APPLE", "1");
--- origsrc/cmake-2.8.0/Source/cmTarget.cxx	2009-11-13 12:32:58.000000000 -0600
+++ src/cmake-2.8.0/Source/cmTarget.cxx	2009-11-24 22:09:36.962175000 -0600
@@ -3036,13 +3036,13 @@ void cmTarget::GetFullNameInternal(const
   outBase += configPostfix?configPostfix:"";
 
   // Name shared libraries with their version number on some platforms.
-  if(const char* version = this->GetProperty("VERSION"))
+  if(const char* soversion = this->GetProperty("SOVERSION"))
     {
     if(this->GetType() == cmTarget::SHARED_LIBRARY && !implib &&
        this->Makefile->IsOn("CMAKE_SHARED_LIBRARY_NAME_WITH_VERSION"))
       {
       outBase += "-";
-      outBase += version;
+      outBase += soversion;
       }
     }
 
--- origsrc/cmake-2.8.0/Source/kwsys/DynamicLoader.cxx	2009-11-13 12:33:01.000000000 -0600
+++ src/cmake-2.8.0/Source/kwsys/DynamicLoader.cxx	2009-11-24 22:11:27.599503100 -0600
@@ -72,7 +72,11 @@ DynamicLoader::GetSymbolAddress(DynamicL
 //----------------------------------------------------------------------------
 const char* DynamicLoader::LibPrefix()
 {
+#if defined(__CYGWIN__)
+  return "cyg";
+#else
   return "lib";
+#endif
 }
 
 //----------------------------------------------------------------------------
--- origsrc/cmake-2.8.0/Source/kwsys/SharedForward.h.in	2009-11-13 12:33:01.000000000 -0600
+++ src/cmake-2.8.0/Source/kwsys/SharedForward.h.in	2009-11-24 22:11:27.623504500 -0600
@@ -254,9 +254,9 @@ static const char kwsys_shared_forward_p
 #endif
 
 /* Windows */
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(__CYGWIN__)
 # if defined(__CYGWIN__)
-#  define KWSYS_SHARED_FORWARD_LDD "cygcheck"
+#  define KWSYS_SHARED_FORWARD_LDD "ldd"
 #  define KWSYS_SHARED_FORWARD_LDD_N 1
 # endif
 # define KWSYS_SHARED_FORWARD_LDPATH "PATH"
--- origsrc/cmake-2.8.0/bootstrap	2009-11-13 12:32:49.000000000 -0600
+++ src/cmake-2.8.0/bootstrap	2009-11-25 02:02:39.965957100 -0600
@@ -624,11 +624,6 @@ cmake_c_flags=${CFLAGS}
 cmake_cxx_flags=${CXXFLAGS}
 cmake_ld_flags=${LDFLAGS}
 
-# Add Cygwin-specific flags
-if ${cmake_system_cygwin}; then
-  cmake_ld_flags="${LDFLAGS} -Wl,--enable-auto-import"
-fi
-
 # Add Carbon framework on Darwin
 if ${cmake_system_darwin}; then
   cmake_ld_flags="${LDFLAGS} -framework Carbon"  
