--- origsrc/cmake-2.8.1/Modules/Platform/CYGWIN.cmake	2010-04-06 09:45:32.000000000 -0500
+++ src/cmake-2.8.1/Modules/Platform/CYGWIN.cmake	2010-04-29 03:18:17.764076400 -0500
@@ -1,9 +1,9 @@
-SET(WIN32 1)
+SET(WIN32 0)
 SET(CYGWIN 1)
 
 SET(CMAKE_SHARED_LIBRARY_PREFIX "cyg")
 SET(CMAKE_SHARED_LIBRARY_SUFFIX ".dll")
-SET(CMAKE_SHARED_MODULE_PREFIX "lib")
+SET(CMAKE_SHARED_MODULE_PREFIX "cyg")
 SET(CMAKE_SHARED_MODULE_SUFFIX ".dll")
 SET(CMAKE_IMPORT_LIBRARY_PREFIX "lib")
 SET(CMAKE_IMPORT_LIBRARY_SUFFIX ".dll.a")
--- origsrc/cmake-2.8.1/Source/cmMakefile.cxx	2010-04-06 09:45:34.000000000 -0500
+++ src/cmake-2.8.1/Source/cmMakefile.cxx	2010-04-29 03:17:22.069890800 -0500
@@ -2344,18 +2344,13 @@ void cmMakefile::AddDefaultDefinitions()
   working, these variables are still also set here in this place, but they
   will be reset in CMakeSystemSpecificInformation.cmake before the platform 
   files are executed. */
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
   this->AddDefinition("WIN32", "1");
   this->AddDefinition("CMAKE_HOST_WIN32", "1");
 #else
   this->AddDefinition("UNIX", "1");
   this->AddDefinition("CMAKE_HOST_UNIX", "1");
 #endif
-  // Cygwin is more like unix so enable the unix commands
-#if defined(__CYGWIN__)
-  this->AddDefinition("UNIX", "1");
-  this->AddDefinition("CMAKE_HOST_UNIX", "1");
-#endif
 #if defined(__APPLE__)
   this->AddDefinition("APPLE", "1");
   this->AddDefinition("CMAKE_HOST_APPLE", "1");
