--- origsrc/cmake-3.0.2/Modules/FindTclsh.cmake	2014-01-16 11:15:08.000000000 -0600
+++ src/cmake-3.0.2/Modules/FindTclsh.cmake	2014-08-28 10:52:09.470930300 -0500
@@ -4,8 +4,6 @@
 #   TCLSH_FOUND = TRUE if tclsh has been found
 #   TCL_TCLSH = the path to the tclsh executable
 #
-# In cygwin, look for the cygwin version first.  Don't look for it later
-# to avoid finding the cygwin version on a Win32 build.
 
 #=============================================================================
 # Copyright 2001-2009 Kitware, Inc.
@@ -20,10 +18,6 @@
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
 
-if(CYGWIN)
-  find_program(TCL_TCLSH NAMES cygtclsh83 cygtclsh80)
-endif()
-
 get_filename_component(TK_WISH_PATH "${TK_WISH}" PATH)
 get_filename_component(TK_WISH_PATH_PARENT "${TK_WISH_PATH}" PATH)
 string(REGEX REPLACE
